### PR TITLE
#162948960 Authenticated users should be able to comment on articles

### PR DESCRIPTION
**Pull request by eduhmik**
### What does this PR do?
This PR allows users who are authenticated into the application to make comments on a specific article

### Description of the PR?
- allow a user to create a new article
- restrict unauthenticated users from commenting
- get all comments on an article

### How to manually test this PR?
- Clone the github repository `git clone https://github.com/andela/ah-fulldeck-frontend` on your local machine.
- Change your working directory to the project folder `cd ah-fulldeck-frontend`
- Checkout to the branch `ft-users-articles-comment-162948960` using the command `git checkout ft-users-articles-comment-162948960`
- Install the required dependencies using `npm install`
- Create a `.env` file with the following values
  ```
  REACT_APP_BASE_URL=
  ```
- Export the environment variables using the command `source .env`
- Run the local development server using `npm start`

### What are the relevant pivotal tracker stories
[#162948960](https://www.pivotaltracker.com/story/show/162948960)

### Screenshots
screenshot 2019-02-20 at 23 00 56

### CODE REVIEW

**Review by Nyakaru — DISMISSED** [review]

File: src/actions/commentsActions.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        dispatch({
+          type: COMMENT_SUCCESS,
+          payload: res.data,
+        });
+      }
+    });
+};
+
+
+// export const updatePassword = (password) => async dispatch => {
```
Comment (victorjambo) [inline src/actions/commentsActions.js]: @eduhmik Commented code

File: src/components/comments/Comment.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/* eslint-disable object-curly-newline */
+/* eslint-disable react/prop-types */
+/* eslint-disable camelcase */
+/* eslint-disable react/prefer-stateless-function */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ComponentButton from './ComponentButton';
+import './Comments.scss';
+
+class Comment extends Component {
```
Comment (victorjambo) [inline src/components/comments/Comment.js]: Make `Comment` a stateless component

File: src/components/comments/CommentsContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      <div>
+        <h1>Comments</h1>
+        <hr />
+        <Comments {...this.props} />
+        {Auth.isAuthenticated && (
+          <form onSubmit={this.onSubmit}>
+            <div className="form-group">
+              <textarea className="form-control" rows="5" id="comment" value={comments} placeholder="Enter your comments here" onChange={this.onChange} errors={errors.comments} />
+            </div>
+            <button type="submit" aria-hidden="true" disabled={this.state.disabled} style={{ height: '40px', width: '30%', float: 'right' }} value="Reset Password" className="btn btn-success btn-block">
```
Comment (victorjambo) [inline src/components/comments/CommentsContainer.jsx]: Avoid inline styles. create a CSS class for it

File: src/components/comments/Comment.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    const { author, created_at, body, author_image } = this.props.comment;
+    return (
+      <div className="container mt-3">
+        <div className="col-md-6 ml-2">
+          <img src={author_image} alt="Avatar" className="avatar" />
+          <p className="d-inline text-muted ml-2">
+            <strong> {author} <span className="ml-2 text-secondary-10" style={{ fontFamily: 'Courier New' }}> {new Date(created_at).toDateString()} </span></strong>
+          </p>
+          <div className="card-text border-0 bg-light text-muted mt-2"> {body} <br /> <br />
+            <div className="col-md-8 ml-2" style={{ float: 'right' }}>
```
Comment (victorjambo) [inline src/components/comments/Comment.js]: Avoid inline styles. create a CSS class for it

**Pull request merged**